### PR TITLE
Add OpImageWrite component count check

### PR DESCRIPTION
**Pull request by sjfricke**
Adds new `VUID-RuntimeSpirv-OpImageWrite-07112` support

This only validates of the `OpImageType` format is not `Unknown`, if it is then `VUID-vkCmdDispatch-None-04115` and `VUID-vkCmdDispatch-OpImageWrite-04469` are used... I was going to add these 2 as well in here, but some other refactoring was needed and bloated this PR

New `CompatibleSpirvImageFormat` generated to map the table in the Spec which SPIR-V image formats go with which `VkFormat`

### CODE REVIEW

**Comment by sjfricke** [comment]
@ncesario-lunarg not sure looking at the `build.gn` script where it would want the SPIRV-Header path as there is nothing explicit for the other files in it (also never worked with the `gn` build system)

File: layers/generated/vk_format_utils.cpp [hunk 1]
```diff
@@ -1927,3 +1928,94 @@ bool FormatHasAlpha(VkFormat format) {
     return FormatHasComponent(format, COMPONENT_TYPE::A);
 }
 
+// Will return the Vulkan format for a given SPIR-V image format value
+// Note: will return VK_FORMAT_UNDEFINED if non valid input
+VkFormat CompatibleSpirvImageFormat(int spirv_image_format) {
```
Comment (ncesario-lunarg) [inline layers/generated/vk_format_utils.cpp:1933]: `int` -> `uint32_t` for `spirv_image_format` type?

File: tests/positive/shaderval.cpp [hunk 2]
```diff
@@ -2934,3 +2934,384 @@ TEST_F(VkPositiveLayerTest, OpTypeStructRuntimeArray) {
     };
     CreateComputePipelineHelper::OneshotTest(*this, set_info, kErrorBit);
 }
+
+TEST_F(VkPositiveLayerTest, StorageImageWriteMoreComponent) {
+    TEST_DESCRIPTION("Test writing to image with less compoents.");
```
Comment (ncesario-lunarg) [inline tests/positive/shaderval.cpp:2939]: `compoents` -> `components` (and similarly below).

**Review by ncesario-lunarg — COMMENTED** [review]
Couple minor nits, but LGTM!

File: build-android/jni/Android.mk [hunk 3]
```diff
@@ -29,6 +29,7 @@ LOCAL_SRC_FILES += $(SRC_DIR)/layers/generated/vk_format_utils.cpp
 LOCAL_C_INCLUDES += $(VULKAN_INCLUDE) \
                     $(LOCAL_PATH)/$(SRC_DIR)/layers/generated \
                     $(LOCAL_PATH)/$(SRC_DIR)/layers \
+                    $(LOCAL_PATH)/$(THIRD_PARTY)/shaderc/third_party/spirv-tools/external/spirv-headers/include \
```
Comment (ncesario-lunarg) [inline build-android/jni/Android.mk:32]: I'm guessing something similar is needed at https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/b48bd2b501be8f63de7279da1d675ec6fef82c3e/BUILD.gn#L271

Comment (sjfricke) [reply build-android/jni/Android.mk:32]: Ok, so there is a `./build-android/secondary/build_overrides/vulkan_validation_layers.gni` that has the paths I am looking for

Comment (ncesario-lunarg) [reply build-android/jni/Android.mk:32]: Ah, good to know...though, it doesn't really make sense for that to be under the `build-android` directory.

**Review by ncesario-lunarg — APPROVED** [review]

File: build-android/jni/Android.mk (hunk 3 above)
Comment (sjfricke) [reply build-android/jni/Android.mk:32]: sorry, for record it is in `build-gn/secondary/build_overrides/vulkan_validation_layers.gni` not `build-android` so make sense :laughing:

**Pull request merged**